### ISSUE TITLE
Dataset Partitioning in Exp3_Kuroshio_forecasting

### ISSUE THREAD

**Issue by Cuiyingzhe** [easylearningscores/Triton_AI4Earth#2]
Thanks for sharing the training and inference codes of Triton. I was particularly impressed by Triton's performance on forecasting Kuroshio eddies when I read the paper. While reviewing the codes under `Triton_AI4Earth/Exp3_Kuroshio_forecasting`, the inconsistency of dataset partitioning between the paper and codes confused me.
The paper says

> ... data from 1993–2020 are used for training, while data from 2021–2024 serve for validation and testing.

However, the code appears to randomly split the full dataset into training, validation, and testing sets.
https://github.com/easylearningscores/Triton_AI4Earth/blob/8b908cd2f562862a04b9cd2a9f844450cd337559/Exp3_Kuroshio_forecasting/dataloader_api/dataloader_kuroshio_256.py#L65-L68
And to the best of my knowledge, in the spatiotemporal forecasting tasks which exhibit strong temporal dependence, the training set should only include samples that chronologically precede those in the test set. This also raises my concern on the inference results of the year 2021 due to the potential information leak. 

I would greatly appreciate any insights someone could share on this matter.

**Comment by easylearningscores** [comment]
After carefully reviewing our experiments, the currently uploaded version of the dataloader is based on the first version of the experiment, which we used for testing model performance. We will upload the version used in the paper next, which is split by publication year. Thank you for your attention and correction.

**Comment by easylearningscores** [comment]
Thank you again for your interest in our work:

The dataloader version we uploaded earlier was the very first one used to test model performance, which is a common practice in spatiotemporal-forecasting research.

Specifically, we feed the model with 10 time steps and predict the next 10 time steps. In the first stage, we did not consider long-term autoregressive inference. Later, in the paper, we demonstrate our method’s inference process, maintaining temporal coherence in spatiotemporal prediction tasks. We used data **before 2021** as the training set and data from **2021 to 2024** as the validation and test sets. We have updated the dataloader used in the paper’s experiments. It is worth emphasizing that no normalization of the dataset is required, so the mean and standard deviation are **0** and **1**.

```python
# Define time range (days since 1950-01-01)
# 1993-01-01 ≈ 15706 (confirmed from first value of time variable in raw data)
# 2020-12-31 ≈ 25931 (365*71 + 18 leap years - 1 day)
# 2023-12-31 ≈ 27027 (end time of raw data)
```

Thank you again for your attention to our work. You are welcome to train and test our method. If you have any further questions, please feel free to continue the discussion ([wuhao2022@mail.ustc.edu.cn](mailto:wuhao2022@mail.ustc.edu.cn)), as this will help us continually improve our research.
